### PR TITLE
Fix worst local variable style offenses

### PR DESCRIPTION
**Pull request by ChillerDragon**
Removed member prefix m_ used for local variables.
Removed all b, c, i hungarian notation prefixes for boolean, const and integers.
Fixed local variables using lower_snake_case instead of UpperCamelCase.
Renamed all ``float wSearch = TextRender()->TextWidth..`` to ``float SearchWidth = ..``.

I spent the entire morning working on https://github.com/ddnet/ddnet/pull/8151. Meaning I was using clang to dump out all local variables that offended the camel case style guide. So I decided to fix the most drastic style violations. This pr is intentionally not fixing ALL style violations to keep the code diff low.

There were usages of the hungarian notation prefixes c, i and b. Introduced by jopsti, robsti and OG magnus him self. While this is correctly following the hungarian notation the rest of the codebase dictates that the ddnet flavor of hungarian notation does not use prefixes for bool, const and integer.

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Review by def- — COMMENTED** [review]
I still think that naming schema is not that important and this will cause conflicts for people (same as all the other renaming changes)

**Comment by ChillerDragon** [comment]
> I still think that naming schema is not that important and this will cause conflicts for people (same as all the other renaming changes)

How do you propose to fix https://github.com/ddnet/ddnet/issues/8142 then? Currently contributors have to follow the naming schema to get merges in and it is audited manually. This is annoying.

I agree with the conflicts thats why I kept the diff minimal and only included the most intense style offenses. That makes the [exclusion regex](https://github.com/ddnet/ddnet/pull/8151/files#diff-bc77863858e7f06147db2e3e2fe04b600ab6eb5b05afc167ebf0bc4d854cc1f1R20) simpler then in the future we could remove style nitpick comments by maintainers from the pull request review process.

```
    value:           '^(p|a|s_|FT_|TB_|s_|ul_|v|[xy]i$|[zijklxyhmrgbacwestnduvqf]$|[dmpwsitcf][xy]$|(ch|skel)[0-2]?$|it$|tw$|dt$|th$|ls$|func$|res$|shader$|len$|maxLength$|length$|offset$|offpos$|result$|bg$|sp$|url$|Tickdelta_legacy$|index$|ctxt$|key$|null$|logger$|LAST_MODIFIED$|GfxFsaaSamples_MouseButton$|teleNr$|target$|id$|hit$|hsl[0-2]?$|rgb[0-2]?$|dir$|tmp$|cData$|sub$|ret$|rendered$|(lower|upper)(16|26|24|32)|size$|wSearch$|bAlreadyHit$|isWeaponCollide$|zerochar$|dist$|sound$|match$|best_skin$|matches$|nohook$|through_cut$|btn$|savedLayers$|l[hw]$|evilz$|sec$|min$|to2$|delay$|m_TileF?Index$|mode$|maxModes$|numModes$|iLogLength$|[xy]Fract$|[xy]Int$|imgg[xy]$|skip$|localPlayer$|fdratio$|[rgbat][0-2]$|[xy][0-3]$|x[rl]$).*'
```

**Comment by def-** [comment]
Styling changes should never be a point of discussion. Just make an automated check if you care for it enough, and then it will be red in CI. If reviewers look for wrong variable names, they should rather spend that time looking for interesting issues.

**Comment by ChillerDragon** [comment]
> Styling changes should never be a point of discussion. Just make an automated check if you care for it enough, and then it will be red in CI. If reviewers look for wrong variable names, they should rather spend that time looking for interesting issues.

Yes I agree. That is why I am working on https://github.com/ddnet/ddnet/pull/8151 but to make the CI pass for the current code base it needs a lot of exceptions. So I thought it would be nice to reduce complexity in the exceptions by fixing those hand picked top offenders. But yea it is not needed. I can also add all of those cases to the exception regex. Just merge or decline fine with me either way.

**Comment by 0xfaulty** [comment]
I think that after the last renaming, those who might have had conflicts (including me) have had many more conflicts than this PR will cause, so it can be applied. So in the future I can be sure that if I also stick to the style, it will not cause conflicts.

**Comment by Learath2** [comment]
In an ideal world, you'd just apply style rules to new commits so we wouldn't have massive style commits but it'd slowly get fixed over time

**Comment by ChillerDragon** [comment]
> In an ideal world, you'd just apply style rules to new commits so we wouldn't have massive style commits but it'd slowly get fixed over time

How to apply style rules to new commits? Checking only the diff to master in CI is possible but annoying. Also running the style check only on the diff locally seems complex. Doing it manually via reviewers nitpicks is annoying.

**Comment by Kaffeine** [comment]
So, you've merged https://github.com/ddnet/ddnet/pull/8019 ("Rename all variables for strict camel-casing of abbreviations" `+5,953−5,953`) which caused **tons** of merge conflicts and introduces inconsistencies in the mods codebase (e.g. the most of mods still use `ClientID`, etc).

And now you're against `+42−42` because
> naming schema is not that important and this will cause conflicts for people

It does not really make sense to stop fixing the few names now.

**Comment by ChillerDragon** [comment]
@Kaffeine I am sorry to hear you had to spend so much time on it. I merged 3 big forks in minutes and rebased one big pr. Make sure to turn on notifications for this issue https://github.com/ChillerDragon/ddnet-meta/issues/1 to not miss instructions for fork maintainers again.

The other pr conflicts could be solved in an automated manner but I feel the only one using automation for that was me. But yes I see your point the merges are a bit inconsitent. If this pr is blocked because its git conflicts can not be solved automatically I can prepare a few shell commands again @def- @Learath2? If that is okay then I could also go with more style offenses than just the worst ones.

Anyways I think it is more important that https://github.com/ddnet/ddnet/pull/8151 is merged quickly because it would protect the master branch from getting new style offenses that are missed in code reviews.

File: src/game/editor/mapitems/layer_game.cpp [hunk 1]
```diff
@@ -38,8 +38,8 @@ void CLayerGame::SetTile(int x, int y, CTile Tile)
 		}
 		CTile nohook = {TILE_NOHOOK};
```
Comment (heinrich5991) [inline src/game/editor/mapitems/layer_game.cpp:39]:
```suggestion
		CTile NoHook = {TILE_NOHOOK};
```
This isn't caught by the CI?

File: src/game/client/gameclient.cpp [hunk 2]
```diff
@@ -2410,18 +2410,18 @@ int CGameClient::IntersectCharacter(vec2 HookPos, vec2 NewPos, vec2 &NewPos2, in
 		if(i == ownId)
 			continue;
 
-		const CClientData &cData = m_aClients[i];
+		const CClientData &Data = m_aClients[i];
```
Comment (heinrich5991) [inline src/game/client/gameclient.cpp:2413]:
```suggestion
		const CClientData &ClientData = m_aClients[i];
```
Or just `Client`.

File: src/game/server/teeinfo.cpp [hunk 3]
```diff
@@ -118,23 +118,23 @@ void CTeeInfo::FromSixup()
 	}
 
 	// find closest match
-	int best_skin = 0;
-	int best_matches = -1;
+	int BestSkin = 0;
+	int BestMatches = -1;
 	for(int s = 0; s < 16; s++)
 	{
 		int matches = 0;
```
Comment (heinrich5991) [inline src/game/server/teeinfo.cpp:125]:
```suggestion
		int Matches = 0;
```
Also not caught?

**Review by heinrich5991 — COMMENTED** [review]
I'm wondering why some very obvious violations aren't caught.

**Comment by ChillerDragon** [comment]
@heinrich5991 this pr is not "all style offenses" it is the "**worst** style offenses" to keep the diff small.

**Comment by ChillerDragon** [comment]
> How was "worst" measured? It caught some but not their immediate neighbors.

Based on my personal taste. Everything that seemed so off that it is worth a diff.

**Comment by ChillerDragon** [comment]
Now that https://github.com/ddnet/ddnet/pull/8151 is merged. I rebased this pr and also removed the exceptions that are no longer needed.

**Review by def- — APPROVED** [review]

**Comment by def-** [comment]
> If that is okay then I could also go with more style offenses than just the worst ones.

Personally I don't think this is important enough to put that much effort in. It is not something that affects players.

**Comment by ChillerDragon** [comment]
> Personally I don't think this is important enough to put that much effort in. It is not something that affects players.

I rather put in effort now to fully automate the style guide than have a review back and forth on every pr.

**Review by Robyt3 — APPROVED** [review]

**Pull request merged**